src/network/wait-online/link.c

```diff
@@ -135,6 +137,50 @@ int link_update_monitor(Link *l) {
                         l->operational_state = s;
         }
 
+        r = sd_network_link_get_required_family_for_online(l->ifindex, &required_family);
+        if (r < 0)
+                ret = log_link_debug_errno(l, r, "Failed to get required address family, ignoring: %m");
+        else if (isempty(required_family))
+                l->required_family = ADDRESS_FAMILY_NO;
+        else {
+                AddressFamily f;
+
+                f = link_required_address_family_from_string(required_family);
+                if (f < 0)
+                        ret = log_link_debug_errno(l, SYNTHETIC_ERRNO(EINVAL),
```

Comment:
When `_from_string()` fails, it returns `-EINVAL`. So, it is not necessary to use `SYNTHETIC_ERRNO(EINVAL)`, but please simply use `f`:
```
ret = log_link_debug_errno(l, f, "Failed to ...");
```